micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java

```diff
@@ -209,6 +210,96 @@ private void monitor(MeterRegistry registry, @Nullable ThreadPoolExecutor tp) {
                 .description("The current number of threads in the pool")
                 .baseUnit(BaseUnits.THREADS)
                 .register(registry);
+
+        monitorRejectedExecutionHandler(registry, tp);
+        monitorThreadFactory(registry, tp);
+    }
+
+    private void monitorThreadFactory(MeterRegistry registry, ThreadPoolExecutor tp) {
+        MonitoredThreadFactory tf = new MonitoredThreadFactory(tp.getThreadFactory());
+        tp.setThreadFactory(tf);
+
+        Gauge.builder("thread.factory.created", tf, MonitoredThreadFactory::getCreated)
+                .tags(tags)
+                .description("The approximate number of threads that create with thread factory")
+                .baseUnit(BaseUnits.THREADS)
+                .register(registry);
+        Gauge.builder("thread.factory.terminated", tf, MonitoredThreadFactory::getTerminated)
+                .tags(tags)
+                .description("The approximate number of threads that is finished execution")
+                .baseUnit(BaseUnits.THREADS)
+                .register(registry);
+        Gauge.builder("thread.factory.running", tf, MonitoredThreadFactory::getRunning)
+                .tags(tags)
+                .description("The approximate number of threads that are started executing, but not terminated")
+                .baseUnit(BaseUnits.THREADS)
+                .register(registry);
+    }
+
+    private void monitorRejectedExecutionHandler(MeterRegistry registry, ThreadPoolExecutor tp) {
+        MonitoredRejectedExecutionHandler mreh = new MonitoredRejectedExecutionHandler(tp.getRejectedExecutionHandler());
+        tp.setRejectedExecutionHandler(mreh);
+
+        Gauge.builder("executor.rejected", mreh, MonitoredRejectedExecutionHandler::getRejectedCount)
```

Comment:
This sounds like a monotonically increasing number. If so, it should be a `FunctionCounter` instead of a `Gauge`.